src/Nethermind/Nethermind.Evm/CodeAnalysis/CodeInfoFactory.cs

```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core.Specs;
+using Nethermind.Evm.EOF;
+
+namespace Nethermind.Evm.CodeAnalysis;
+
+public static class CodeInfoFactory
+{
+    public static ICodeInfo CreateCodeInfo(byte[] code, IReleaseSpec spec)
+    {
+        CodeInfo codeInfo = new(code);
+        return spec.IsEip3540Enabled && EvmObjectFormat.IsValidEof(code, out EofHeader? header)
```

Comment:
Aren't we calling `IsValidEoF` too much? I think the idea is that if it is a contract code, then it is validated during `CREATE` and we don't have to validate it again when loading it back to execute.

Reply:
code that is sent with a transaction can be EOF, so when we create a codeinfo for that, we need to check that it is EOF. Or maybe i am missing something